test-integration/DepositZap.js

```diff
@@ -0,0 +1,347 @@
+const { ethers } = require("hardhat");
+const { bytes32 } = require("../utils/helpers");
+const timeMachine = require("ganache-time-traveler");
+const { WHALE_POOLS } = require("../utils/constants");
+const {
+  acquireToken,
+  console,
+  tokenAmountToBigNumber,
+  deployAggregator,
+  generateContractAddress,
+} = require("../utils/helpers");
+
+const DAI_ADDRESS = "0x6B175474E89094C44Da98b954EedeAC495271d0F";
+const USDC_ADDRESS = "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48";
+const TETHER_ADDRESS = "0xdAC17F958D2ee523a2206206994597C13D831ec7";
+const CURVE_3CRV_ADDRESS = "0x6c3F90f043a72FA612cbac8115EE7e52BDe6E490";
+const USDC_AGG_ADDRESS = "0x8fFfFfd4AfB6115b954Bd326cbe7B4BA576818f6";
+
+/* ************************ */
+/* set DEBUG log level here */
+/* ************************ */
+console.debugging = false;
+/* ************************ */
+
+async function deployProxy(logic, proxyAdmin, initData) {
+  const TransparentUpgradeableProxy = await ethers.getContractFactory(
+    "TransparentUpgradeableProxy"
+  );
+  const proxy = await TransparentUpgradeableProxy.deploy(
+    logic.address,
+    proxyAdmin.address,
+    initData
+  );
+  await proxy.deployed();
+  return proxy;
+}
+
+describe("Contract: IndexToken", () => {
+  // signers
+  let deployer;
+  let oracle;
+  let adminSafe;
+  let emergencySafe;
+  let randomUser;
+
+  // Mainnet ERC20s
+  let curve3Crv;
+  let dai;
+  let usdc;
+  let tether;
+
+  // protocol contracts
+  let tvlAgg;
+  let oracleAdapter;
+  let mApt;
+  let addressRegistry;
+  let proxyAdmin;
+
+  // system under test
+  let indexToken;
+  let depositZap;
+
+  before(async () => {
+    [deployer, oracle, adminSafe, emergencySafe, randomUser] =
+      await ethers.getSigners();
+  });
+
+  // use EVM snapshots for test isolation
+  let testSnapshotId;
+  let suiteSnapshotId;
+
+  beforeEach(async () => {
+    const snapshot = await timeMachine.takeSnapshot();
+    testSnapshotId = snapshot["result"];
+  });
+
+  afterEach(async () => {
+    await timeMachine.revertToSnapshot(testSnapshotId);
+  });
+
+  before(async () => {
+    const snapshot = await timeMachine.takeSnapshot();
+    suiteSnapshotId = snapshot["result"];
+  });
+
+  after(async () => {
+    await timeMachine.revertToSnapshot(suiteSnapshotId);
+  });
+
+  before("Setup Chainlink Aggregator", async () => {
+    const paymentAmount = tokenAmountToBigNumber("1", 18); // LINK
+    const maxSubmissionValue = tokenAmountToBigNumber("1", "20");
+    const tvlAggConfig = {
+      paymentAmount, // payment amount (price paid for each oracle submission, in wei)
+      minSubmissionValue: 0,
+      maxSubmissionValue,
+      decimals: 8, // decimal offset for answer
+      description: "TVL aggregator",
+    };
+    tvlAgg = await deployAggregator(
+      tvlAggConfig,
+      oracle.address,
+      deployer.address, // oracle owner
+      deployer.address // ETH funder
+    );
+  });
+
+  before("Deploy and Setup Address Registry", async () => {
```

Comment:
We should keep in mind that once we remove the address registry dependency from the index token we can remove this entire set up.